test/stdlib/TestJSONEncoder.swift

```diff
@@ -92,6 +92,17 @@ class TestJSONEncoder : TestJSONEncoderSuper {
     _testRoundTrip(of: employee)
   }
 
+  func testEncodingTopLevelNullableType() {
+    // EnhancedBool is a type which encodes either as a Bool or as nil.
+    _testEncodeFailure(of: EnhancedBool.true)
+    _testEncodeFailure(of: EnhancedBool.false)
+    _testEncodeFailure(of: EnhancedBool.fileNotFound)
```

Comment:
Hehe, I told @phausler I was glad to finally be able to use this somewhere in real code.